src/test/java/it/pagopa/transactions/services/v1/CircuitBreakerTest.java

```
@@ -54,9 +64,315 @@ class CircuitBreakerTest {
+                                .elements(),
+                        Spliterator.ORDERED
+                ),
+                false
+        )
+                .map(json -> {
+                    String exceptionName = json.asText();
+                    return Arguments.of(
+                            Optional
+                                    .ofNullable(exceptionMapper.get(exceptionName))
+                                    .orElseThrow(
+                                            () -> new RuntimeException(
+                                                    "No mapping found for exception: %s".formatted(exceptionName)
+                                            )
+                                    ),
+                            retryInstanceName
+                    );
+                }
+
+                );
+    }
+
+    private static Stream<Arguments> getIgnoredExceptionForNewTransactionRetry() {
+        return getIgnoredExceptionsForRetry("newTransaction");
+    }
+
+    private static Stream<Arguments> getIgnoredExceptionForGetTransactionInfoRetry() {
+        return getIgnoredExceptionsForRetry("getTransactionInfo");
+    }
+
+    private static Stream<Arguments> getIgnoredExceptionForRequestTransactionAuthorizationRetry() {
+        return getIgnoredExceptionsForRetry("requestTransactionAuthorization");
+    }
+
+    private static Stream<Arguments> getIgnoredExceptionForUpdateTransactionAuthorizationRetry() {
+        return getIgnoredExceptionsForRetry("updateTransactionAuthorization");
+    }
+
+    private static Stream<Arguments> getIgnoredExceptionForCancelTransactionRetry() {
+        return getIgnoredExceptionsForRetry("cancelTransaction");
+    }
+
+    private static Stream<Arguments> getIgnoredExceptionForAddUserReceiptRetry() {
+        return getIgnoredExceptionsForRetry("addUserReceipt");
+    }
+
+    @ParameterizedTest
+    @MethodSource("getIgnoredExceptionForNewTransactionRetry")
+    @Order(0)
```

Why is an explicit order needed? Is it to control the circuit breaker behaviour?

---

Yes, I want to run first all tests that will not open circuit breaker (ignored exceptions) then the last one that tests circuit breaker opening :)